addon/commands/unindent-list-command.ts

```diff
@@ -0,0 +1,87 @@
+import Command from "@lblod/ember-rdfa-editor/commands/command";
+import ModelSelection from "@lblod/ember-rdfa-editor/model/model-selection";
+import Model from "@lblod/ember-rdfa-editor/model/model";
+import {MisbehavedSelectionError, SelectionError} from "@lblod/ember-rdfa-editor/utils/errors";
+import ModelNode from "@lblod/ember-rdfa-editor/model/model-node";
+import ModelElement from "@lblod/ember-rdfa-editor/model/model-element";
+import {listTypes} from "@lblod/ember-rdfa-editor/model/util/constants";
+import ModelNodeFinder from "@lblod/ember-rdfa-editor/model/util/model-node-finder";
+export default class UnindentListCommand extends Command {
+  name: string = "unindent-list";
+  constructor(model: Model) {
+    super(model);
+  }
+
+  execute(selection: ModelSelection = this.model.selection): void {
+    if(!ModelSelection.isWellBehaved(selection)) {
+
+      throw new MisbehavedSelectionError();
+    }
+    //start of selection
+    const anchorNode = selection.lastRange.start.parent;
+    //end of selection
+    const focusNode = selection.lastRange.end.parent;
+
+    //get all nodes in selection that are li
+    const listNodesIterator = selection.findAllInSelection({
+      filter: ModelNode.isModelElement,
+      predicate: (node: ModelElement) => node.type === "li"
+    });
+
+
+    //if there are no li in selection return error
+    if (!listNodesIterator) {
+      throw new SelectionError('The selection is not in a list');
+    }
+
+    //iterate over all found li elements
+    for (const li of listNodesIterator){
+      //check that the li is nested
+      if(li.findAncestor(node => ModelNode.isModelElement(node) && node.type=='li')){
+        //before:
+        //ul    great-grandparent
+        // li   grandparent
+        //  ul  parent
+        //   li node
+
+        //after:
+        //ul    great-grandparent
+        // li   grandparent
+        // li   node
+
+        const node=li;
+        const parent=li.findAncestor(node => ModelNode.isModelElement(node) && listTypes.has(node.type), false) as ModelElement;
+        const grandParent=parent?.findAncestor(node => ModelNode.isModelElement(node) && listTypes.has(node.type), false) as ModelElement;
+        const greatGrandParent=grandParent?.findAncestor(node => ModelNode.isModelElement(node) && listTypes.has(node.type), false) as ModelElement;
+
+        if(node && parent && grandParent && greatGrandParent){
+          //remove node
+          const nodeIndex=node.index!;
+          parent.removeChild(node);
+
+          //remove parent if node is only child
+          if(parent.length==0){
+            if(parent.index){
+              grandParent.addChild(node, parent.index);
+              this.model.write();
```

Comment:
this shouldn't be necessary here since you are writing the whole model at the end already

Reply:
this is an escape clause it returns if this "if" is reached hence i am writing the model. maybe there is a way to break out of nested if inside a loop in ts im am unaware of though?

Reply:
oh i forgot you can break =) in for loops